sandman/source/schedule.cpp

```diff
@@ -26,22 +21,18 @@
 // Types
 //
 
-// A schedule event.
-struct ScheduleEvent
+class Schedule 
 {
-	// Read a schedule event from JSON. 
-	//
-	// p_EventObject:	The JSON object representing the event.
-	//	
-	// Returns:		True if the event was read successfully, false otherwise.
-	//
-	bool ReadFromJSON(rapidjson::Value::ConstObject const& p_EventObject);
-	
-	// Delay in seconds before this entry occurs (since the last).
-	unsigned int	m_DelaySec;
-	
-	// The control action to perform at the scheduled time.
-	ControlAction	m_ControlAction;
+    public:
+        Schedule() = default;
+        void AddEvent(const ScheduleEvent& p_event);
+        bool IsEmpty();
+        size_t GetNumEvents() const;
+        std::vector<ScheduleEvent> GetEvents() const;
```

Comment:
`std::vector<ScheduleEvent> const&`

Reply:
will need to fix this line before we can do this:
l_Control = l_Event.m_ControlAction.GetControl()

Reply:
Yeah, we discussed this and one option for fixing this would be to make it so that the controls didn't use handles anymore and instead did a look up every time. This would be performant with a map or something like that.